.github/workflows/tests_core.yml

```
@@ -0,0 +1,27 @@
+name: Core Tests
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  packages:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.x"
+
+    - name: Core Tests
+      run: >
+        python -m pip install .
+        && python -m pip install pytest
+        && python -m pytest -rxs tests/test_erddapy.py
```

Maybe split out the install into a separate step? Makes it a bit easier to figure out which step failed.

Should these also be tested against a matrix of versions?

```suggestion

    - name: Install
      run: >
        python -m pip install .
        && python -m pip install pytest
    
    - name: Core Tests
      run:  python -m pytest -rxs tests/test_erddapy.py
```

---

> Should these also be tested against a matrix of versions?

It is not needed IMO b/c we do repeat these tests in the full test matrix. The goal of this one is to catch issues in a non-conda env (like the missing httpx! :fearful:) and avoid regressions with the optional imports.